src/helpers/domain.js

```diff
@@ -26,8 +26,13 @@ export default {
     const { horizontal } = props;
     const ensureZero = (domain) => {
       const isDependent = (axis === "y" && !horizontal) || (axis === "x" && horizontal);
-      const zeroDomain = isDependent ? [Math.min(...domain, 0), Math.max(... domain, 0)]
-      : domain;
+      const min = Collection.containsDates(domain) ?
+        Helpers.retainDate(Math.min(...domain, 0)) :
```

Comment:
if all `retainDate` does is `new Date(val)` why not just do that here? When we were chatting about extracting helper methods I was imagining `getMaxValue(arr)` which would do all this date checking and return the correct value type. If you end up going that route, the new methods should be added to `util/collection` since they operate on arrays. I thought this would be happening in may more places though, so if you just want to swap out `retainDate(val)` for `new Date(val)` that's fine too